src/ffi/models/index.js

```diff
@@ -0,0 +1,69 @@
+import ref from 'ref';
+import StructType from 'ref-struct';
+import Enum from 'enum';
+import {
+  refString,
+  usize,
+  u8Arr,
+  u8Pointer,
+  bool
+} from './ref_types';
+
+export const AppKeys = StructType({
+  owner_key: u8Arr,
+  enc_key: u8Arr,
+  sign_pk: u8Arr,
+  sign_sk: u8Arr,
+  enc_pk: u8Arr,
+  enc_sk: u8Arr
+});
+
+export const AppExchangeInfo = StructType({
+  id: refString,
+  scope: u8Pointer,
+  scope_len: usize,
+  scope_cap: usize,
+  name: refString,
+  vendor: refString
+});
+
+export const Permission = Enum({
```

Comment:
Has this been compiled? doing `Enum({` without a `new` fails for me here with a `TypeError: Cannot call a class as a function` error...